### PR TITLE
New algorithms for exporting and importing bookmarks

### PR DESCRIPTION
**Pull request by nyalldawson**
These algorithms allow users to create a bunch of new bookmarks corresponding to the features from a layer, or export bookmark extents to a new polygon layer

### CODE REVIEW

File: src/analysis/processing/qgsbookmarkalgorithms.cpp [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+QString QgsBookmarksToLayerAlgorithm::name() const
+{
+  return QStringLiteral( "bookmarkstolayer" );
+}
+
+QString QgsBookmarksToLayerAlgorithm::displayName() const
+{
+  return QObject::tr( "Convert bookmarks to layer" );
```
Comment (nirvn) [inline src/analysis/processing/qgsbookmarkalgorithms.cpp]: I'd suggest "Convert _spatial_ bookmarks to layer" as a better display name.

File: src/analysis/processing/qgsbookmarkalgorithms.cpp [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+QString QgsBookmarksToLayerAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeneral" );
+}
+
+QString QgsBookmarksToLayerAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm creates a new layer containing polygon features for stored bookmarks.\n\n"
```
Comment (nirvn) [inline src/analysis/processing/qgsbookmarkalgorithms.cpp]: stored bookmarks -> stored spatial bookmarks.

File: src/analysis/processing/qgsbookmarkalgorithms.cpp [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                      "The export can be filtered to only bookmarks belonging to the current project, to all user bookmarks, or a combination of both." );
+}
+
+QString QgsBookmarksToLayerAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Converts stored bookmarks to a polygon layer." );
```
Comment (nirvn) [inline src/analysis/processing/qgsbookmarkalgorithms.cpp]: Same wording suggestion here.

File: src/analysis/processing/qgsbookmarkalgorithms.cpp [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+QString QgsLayerToBookmarksAlgorithm::name() const
+{
+  return QStringLiteral( "layertobookmarks" );
+}
+
+QString QgsLayerToBookmarksAlgorithm::displayName() const
+{
+  return QObject::tr( "Convert layer to bookmarks" );
```
Comment (nirvn) [inline src/analysis/processing/qgsbookmarkalgorithms.cpp]: +spatial

File: src/analysis/processing/qgsbookmarkalgorithms.cpp [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+QString QgsLayerToBookmarksAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeneral" );
+}
+
+QString QgsLayerToBookmarksAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm creates bookmarks corresponding to the extent of features contained in a layer." );
```
Comment (nirvn) [inline src/analysis/processing/qgsbookmarkalgorithms.cpp]: +spatial

File: src/analysis/processing/qgsbookmarkalgorithms.cpp [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+QString QgsLayerToBookmarksAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Converts feature extents to stored bookmarks." );
```
Comment (nirvn) [inline src/analysis/processing/qgsbookmarkalgorithms.cpp]: +spatial

**Review by nirvn — APPROVED** [review]
Fantastic. My comments only refer to one wording change (i.e. bookmark -> spatial bookmark).

**Comment by nyalldawson** [comment]
Thanks @nirvn, addressed

**Pull request merged**